### PR TITLE
Test API improvements: pumpAndSettle, autogenerate pointer IDs

### PR DESCRIPTION
**Pull request by Hixie**
I have an e-mail to flutter-dev ready to go.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @abarth

**Comment by Hixie** [comment]
Fixes https://github.com/flutter/flutter/issues/8186

File: packages/flutter_test/lib/src/test_pointer.dart [hunk 1]
```diff
@@ -15,13 +15,34 @@ export 'dart:ui' show Point;
...
+  }
+
+  TestPointer._(this.pointer);
+
+  /// The next available pointer identifier.
+  ///
+  /// This is the default pointer identifier that will be used the next time the
+  /// [new TestPointer] constructor is called without an explicit pointer
+  /// identifier.
+  static int nextPointer = 1;
```
Comment (abarth) [inline packages/flutter_test/lib/src/test_pointer.dart]: This creates cross-test dependencies.  Maybe put the counter on the WidgetTester or the test bindings?

**Comment by abarth** [comment]
LGTM

**Pull request merged**